SilverSolver/assignment1/Q2.py

```diff
@@ -1,40 +1,23 @@
-from collections import deque
-
 class single_linked_node:
 
     def __init__(self, value=None, child=None):
         self.child = child
         self.value = value
 
-    def __getitem__(self, k, one_pass=True):
-        list_len = 0
-        visited_nodes = deque(maxlen=-k + 1)
+    def __getitem__(self, k):
+        current_pos = 0
+        k_th_back_node = self
         current_node = self
         while (current_node):
-            # if we want optimize by time and do one_pass through the
-            # linked list, we should keep k last visited nodes
```

Comment:
I agree with you Laida, doing only one pass is not an optimization if the single pass does twice more work. The one pass is kust an alternative, thanks for pointing it